web-common/src/features/dashboards/dimension-table/ExportDimensionTableDataButton.svelte

```diff
@@ -46,33 +46,12 @@
     }
   });
 
-  $: scheduledReportsQueryArgsJson = JSON.stringify({
-    instanceId: get(runtime).instanceId,
-    metricsViewName: metricViewName,
-    dimension: {
-      name: $dashboardStore.selectedDimensionName,
-    },
-    measures: $dashboardStore.selectedMeasureNames.map((name) => ({
-      name: name,
-    })),
-    timeRange: {
-      start: $timeControlStore.timeStart,
-      end: $timeControlStore.timeEnd,
-    },
-    comparisonTimeRange: {
-      start: $timeControlStore.comparisonTimeStart,
-      end: $timeControlStore.comparisonTimeEnd,
-    },
-    sort: [
-      {
-        name: $dashboardStore.leaderboardMeasureName,
-        desc: $dashboardStore.sortDirection === SortDirection.DESCENDING,
-        type: getQuerySortType($dashboardStore.dashboardSortType),
-      },
-    ],
-    filter: $dashboardStore.filters,
-    offset: "0",
-  });
+  $: scheduledReportsQueryArgsJson = getDimensionTableExportArgs(
+    metricViewName,
+    $dashboardStore,
+    $timeControlStore,
+    $metaQuery.data
+  );
```

Comment:
Could be cleaner if the function signature were `getDimensionTableExportArgs(ctx)` given all the stores can be accessed from the state management context.

Reply:
Good point